src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,296 @@
+import logging
+from itertools import groupby
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import numpy
+import numpy as np
+import rasterio
+import rasterio.features
+from pystac import Item
+from rasterio import Affine
+from rasterio.crs import CRS
+from rasterio.warp import transform_geom
+from shapely.geometry import mapping, shape
+from shapely.geometry.multipolygon import MultiPolygon
+from shapely.geometry.polygon import Polygon
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_SCALE = 1
+DEFAULT_PRECISION = 3
+
+
+def update_geometry_from_asset_footprint(
+    item: Item,
+    *,
+    asset_names: Optional[List[str]] = None,
+    scale: int = DEFAULT_SCALE,
+    precision: int = DEFAULT_PRECISION,
+    densification_factor: Optional[int] = None,
+    simplify_tolerance: Optional[float] = None,
+    no_data: Optional[int] = None,
+) -> bool:
+    """
+    Accepts an Item and an optional list of asset names within that item, and updates
+    the geometry of that Item in-place with the data footprint derived from the first
+    of the assets that exists in the Item.
+
+    Two important operations during this calculation are the densification of the
+    footprint in the native CRS and simplification of the footprint after reprojection.
+    If the initial low-vertex polygon in the native CRS is not densified, this can result
+    in a reprojected polygon that does not accurately represent the data footprint. For
+    example, a MODIS asset represented by a rectangular 5 point Polygon in a sinusoidal
+    projection will reproject to a parallelogram in EPSG 4326, when it would be more
+    accurately represented by a polygon with two parallel sides and two curved sides.
+    The difference between these representations is even greater the further away from
+    the meridian and equator the asset is located.
+
+    After reprojection to EPSG 4326, the footprint may have more points than desired.
+    This can be simplified to a polygon with fewer points that maintain a minimum distance
+    between each point.
```

Comment:
Update to match corrected description of simplification (as in the `simplify_tolerance` description in Args below).

Reply:
IIUC, simplification does not impart a minimum distance between points; it looks at the distance between a point and the new geometry should that point be removed. See [here].(https://shapely.readthedocs.io/en/stable/manual.html#object.simplify)

The sentence should read something like, "This can be simplified to a polygon with fewer points that maintain a minimum distance _to the original geometry_."